config/rbac/role.yaml

```diff
@@ -26,3 +26,23 @@ rules:
   - get
   - patch
   - update
+- apiGroups:
+    - metal3.io
+  resources:
+    - baremetalhosts
+  verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+- apiGroups:
+    - metal3.io
+  resources:
+    - baremetalhosts/status
+  verbs:
+    - get
+    - patch
+    - update
```

Comment:
Whenever you run the kubebuilder generation again, this will be lost, if you do not add it in the kubebuilder tags.

Reply:
It should be added here : https://github.com/metal3-io/hardware-classification-controller/blob/7c6e93733b1c94dc3a974e30b915bd2180333496/controllers/hardwareclassificationcontroller_controller.go#L36

Reply:
Will add there. thanks @maelk

Reply:
@sharajat  I agree with @maelk.  We are only fetching BaremetalHosts. Please keep only get and list action